### PR TITLE
Enable deployments for CARTSv3

### PR DESCRIPTION
**Pull request by bodnarbm**
# Description

Automates deployments and cypress testing on the main, val, and production environments for CARTS v3. ~Dev branches are out of scope for this PR and will be added later when a workaround for the default branch issue is figured out.~ Dev branches are now supported with this PR pending #2479 merging first.

Also added the application endpoint to the job summary for the deploy job to make finding the endpoint easier.

image

Addresses [MDCT-299](https://qmacbis.atlassian.net/browse/MDCT-299).

## How to test

This was tested on this branch using a prior commit that triggered deployments on this branch.

Deployment: https://github.com/CMSgov/cms-carts-seds/actions/runs/2332826198

Deployment Support: https://github.com/CMSgov/cms-carts-seds/actions/runs/2332826197

## Dependencies

Moved cypress dependencies to be with the rest of cypress.

# Get it done

Now that the PR is open this is what happens next

## Code authors checklist

- [x] I have performed a self-review of my code
- [x] I have added thorough tests. [Testing Doc](https://qmacbis.atlassian.net/wiki/spaces/CM/pages/2914025525/Test+Suite+and+Testing+Research)
- [x] Necessary analytics were added, or no new analytics were required
- [x] I have made corresponding changes to the documentation
- [x] I have assigned the PR to myself

## Reviewers Checklist (Two different people)

- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

## Assignee

- [ ] I have closed the PR after the review and necessary changes (squashing preferred)

### CODE REVIEW

**Review by BearHanded — DISMISSED** [review]

**Review by gmrabian — APPROVED** [review]
cool with me

File: tests/cypress/support/commands.js [hunk 1]
```diff
@@ -9,13 +11,40 @@ before(() => {
...
+        credentials = stateUser;
+        break;
+      default:
+        throw new Error("Provided userType not recognized.");
+    }
+  } else {
+    throw new Error("Must specify either userType or userCredentials.");
+  }
+  cy.xpath(emailForCognito).type(credentials.email);
+  cy.xpath(passwordForCognito).type(credentials.password);
```
Comment (braxex) [inline tests/cypress/support/commands.js:47]: This section in particular is amazing.

**Review by braxex — APPROVED** [review]
LGTM

File: tests/cypress/support/commands.js (hunk 1 above)
Comment (bodnarbm) [reply tests/cypress/support/commands.js:47]: 😄 Stolen from MCR / QMR.

Comment (braxex) [reply tests/cypress/support/commands.js:47]: lol i wrote it 😏

**Pull request merged**